Include/internal/pycore_bitutils.h

```diff
@@ -131,8 +131,43 @@ _Py_popcount32(uint32_t x)
 }
 
 
+// Return the index of the most significant 1 bit in 'x'. This is the smallest
+// integer k such that x < 2**k. Equivalent to floor(log2(x)) + 1 for x != 0.
+static inline int
+_Py_bit_length(unsigned long x)
+{
+    if (x == 0) {
+        return 0;
+    }
+#if (defined(__clang__) || defined(__GNUC__))
+    // __builtin_clzl() is available since GCC 3.4.
+    // Undefined behavior for x == 0.
+    return sizeof(unsigned long) * 8 - __builtin_clzl(x);
+#elif defined(_MSC_VER)
+    Py_BUILD_ASSERT(sizeof(unsigned long) <= 4);
```

Comment:
What is the purpose of this assertion? The documentation says:

```
unsigned char _BitScanReverse(
   unsigned long * Index,
   unsigned long Mask
);
```

The function accepts an unsigned long.

Reply:
The function accepts `unsigned long` but is documented to search 32 bits. So if there's a platform with `_MSC_VER` but `sizeof(unsigned long) > 4` our `_Py_Bit_Length(unsigned long)` would not work for values >= 2**32. The assertion is that such a platform does not exist.